sdk/core/azure-core/src/main/java/com/azure/core/util/polling/StatusCheckPollingStrategy.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.implementation.TypeUtil;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+/**
+ * Fallback polling strategy that doesn't poll but exits successfully if no other polling are detected
+ * and status code is 2xx.
+ *
+ * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+ *            should be kept
+ * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+ *            body should be kept
+ */
+public class StatusCheckPollingStrategy<T, U> implements PollingStrategy<T, U> {
+
+    private final ClientLogger logger = new ClientLogger(StatusCheckPollingStrategy.class);
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        return Mono.just(true);
```

Comment:
Shouldn't this check the status of the initialResponse to see if the status is 2xx?

Reply:
Added a status check.